core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/RebindIteration.java

```diff
@@ -687,8 +689,13 @@ protected void manageTheObjects() {
         Set<String> oldEntities = Sets.newLinkedHashSet(entityManager.getEntityIds());
         for (Entity entity: rebindContext.getEntities()) {
             ManagementTransitionMode oldMode = updateTransitionMode(entityManager, entity);
-            if (oldMode!=null)
+            if (oldMode!=null) {
                 oldEntities.remove(entity.getId());
+            }
+
+            if (ServiceStateLogic.getExpectedState(entity) == Lifecycle.STARTING || ServiceStateLogic.getExpectedState(entity) == Lifecycle.STOPPING) {
```

Comment:
I think this call needs to be guarded so it's not called by a HOT_STANDBY Brooklyn server (i.e. a read-only view) etc. We only want to set this if this entity is in `BrooklynObjectManagementMode.MANAGED_PRIMARY`.

(Note that is not the `oldMode`; it is the new mode that we care about).

I therefore wonder if there is a better place for this code. I'll dig around and see.